projects/libs/core-library-typescript/src/subscription-manager/subscription-manager.ts

```diff
@@ -16,7 +26,58 @@ export class SubscriptionManager {
     this.contractProxy = new ERCBillingContractProxy(sidechainConnector, this.config.ethereumContractAddress);
   }
 
-  async getSubscriptionStatus(subscriptionKey: string) {
-    return await this.contractProxy.getSubscription(subscriptionKey);
+  async isSubscriptionValid(subscriptionKey: string): Promise<boolean> {
+    const subscriptionData =  await this.contractProxy.getSubscriptionData(subscriptionKey);
+
+    const now = new Date();
+
+    const rateData = this.getCurrentSubscriptionRateByProfile(subscriptionData.profile, now);
+
+    if (rateData == undefined) {
+      logger.info(`Could not find rate settings for subscription profile ${subscriptionData.profile}`);
+      return false;
+    }
+
+    const beginningOfThisMonth = new Date(`${now.toISOString().substr(0, 7)}-01`).getTime();
+
+    let minimumRequiredFee: number;
+
+    if (subscriptionData.startTime <= beginningOfThisMonth) {
```

Comment:
Should test that startTime is in the past. (Not critical because the contract handles it already)